project3/coordinator/registrar.h

```diff
@@ -0,0 +1,52 @@
+/**
+ * @file Registrar to be utilized by the coordinator.
+ */
+#ifndef PROJECT3_REGISTRAR_H
+#define PROJECT3_REGISTRAR_H
+#include "connected_participants.h"
+#include <memory>
+
+namespace coordinator::registrar {
+using namespace connected_participants;
+/**
+ * @class Handles registration/de-registration of participants.
+ */
+class Registrar {
+ private:
```

Comment:
(minor) Usually, we put `public` before `private`, as it makes it easier for readers.